kolibri/core/assets/src/components.js

```diff
@@ -149,6 +73,132 @@ var CrudCollection = Mn.CollectionView.extend({
 });
 
 
+var CrudAddItem = Mn.ItemView.extend({
+    // Template should either be the html *or* (as it is here) be a *function* that returns the rendered html.
+    // If it's a function, it's passed an object which is the "serialized" model of the view instance.
+    // In other words, if the view instance is given a model attribute when instantiated, then the argument
+    // to this function will be a copy of that model's .attributes hash.
+    template: function() {
+        var html =
+            '{{#unless valid}}<span>Invalid Input!</span><br />{{/unless}}' +
+            '{{#each create}}' +
+                '<input data-attr="{{ name }}" type="text" placeholder="{{ name }}" value="{{ default }}" /><br />' +
+            '{{/each}}' +
+            '<button class="create flat-button">Create</button>';
+        return Handlebars.compile(html)({create: this.create, valid: this.valid});
```

Comment:
I wanted to be able to distinguish between fields which should be displayed on a creation form vs fields that should be editable in the parent CrudView's config options -- hence `create` vs the implied `edit`. But perhaps better would be `creatable` & `editable` (or `creatable_fields` & `editable_fields`).